libs/angular-google-charts/src/lib/services/data-table.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+
+import { Column, Row } from '../components/chart-base/chart-base.component';
+import { Formatter } from '../types/formatter';
+
+@Injectable({ providedIn: 'root' })
+export class DataTableService {
+  public create(data: Row[] | undefined, columns?: Column[], formatters?: Formatter[]): google.visualization.DataTable | undefined {
+    if (data == null) {
+      return undefined;
+    }
+
+    let firstRowIsData = true;
+    if (columns != null) {
+      firstRowIsData = false;
+    }
+
+    const dataTable = google.visualization.arrayToDataTable(this.getDataAsTable(data, columns), firstRowIsData);
+    if (formatters) {
+      this.applyFormatters(dataTable, formatters);
+    }
+
+    return dataTable;
+  }
+
+  private getDataAsTable(data: Row[], columns: Column[] | undefined): (Row | Column[])[] {
+    if (columns) {
+      return [columns, ...data];
+    } else {
+      return data;
+    }
+  }
```

Comment:
Note that dashboard.component.ts has nearly identical code, and should also be refactored.
The simplest approach is for Dashboard to call DTService.create() without formatters.
A future PR will likely add formatters to dashboard.